### PR TITLE
Refactor AST to use meta parameterization

### PR DESCRIPTION
**Pull request by enetsee**
Changes for #105

### CODE REVIEW

**Comment by seantalts** [comment]
Tried to fix the merge conflicts; PTAL and make sure I didn't mess anything up!

**Comment by VMatthijs** [comment]
Oh this is great! Very nice that you had time to make this refactor. Wanted to do it myself but have been so busy.

**Comment by seantalts** [comment]
Yeah we can 😂 Just pushed a commit to remove it.

**Comment by VMatthijs** [comment]
@seantalts ,  any idea what is happening with Jenkins here?

**Comment by seantalts** [comment]
Tagging in @serban-nicusor-toptal

**Comment by serban-nicusor-toptal** [comment]
Should be just fine now.

**Comment by seantalts** [comment]
Thank you! Merged :D

File: lib/Ast.ml [hunk 1]
```diff
@@ -61,30 +61,41 @@ type 'e expression =
...
 [@@deriving sexp, hash, compare, map]
 
+type 'm expr_with = {expr: 'm expr_with expression; emeta: 'm}
+[@@deriving sexp, compare, map, hash]
+
 (** Untyped expressions, which have location_spans as meta-data *)
-type untyped_expression =
-  { expr_untyped: untyped_expression expression
-  ; expr_untyped_loc: Mir.location_span sexp_opaque [@compare.ignore] }
+type located_meta = {loc: Mir.location_span sexp_opaque [@compare.ignore]}
```
Comment (seantalts) [inline lib/Ast.ml]: Hey, just looking at this - we don't actually need to use a record for located_meta, it's just a span. Can I remove this layer of indirection?

Comment (seantalts) [reply lib/Ast.ml]: Or if you guys think we should have them always be a record we'd want to change the MIR to match I think. Right now it's just defined like this: https://github.com/stan-dev/stanc3/blob/master/lib/Mir.ml#L165

Comment (enetsee) [reply lib/Ast.ml]: I think using the `location_span` directly makes sense given we are unlikely to attach more meta data at this point.

**Pull request merged**